File: MPOO/clase-21082017/factorial/swift/factorial.swift
class Factorial{
	var total:Int
	
	// Este es el constructor en swift
	init(){
		self.total = 1
	}
	
	func calcula(valor:Int) -> Int{
		var n = valor
		while n > 1{
			total = total * n
			n = n - 1
		}
		return total
	}
}

var f = Factorial()
print(f.calcula(valor:5))

File: MPOO/clase-21082017/factorial/java/Factorial.java
public class Factorial{
    int total;

    // El constructor en Java
    public Factorial(){
        total = 1;
    }


    int calcula(int n){
        while(n > 1){
            total = total * n;
            n = n - 1;
        }
        return total;
    }

}

